apps/cfp_review/base.py

```diff
@@ -311,6 +312,13 @@ def log_and_close(msg, next_page, proposal_id=None):
                 next_id=next_id,
             )
 
+        # Create any new tags that are needed
+        new_tags = [t for t in prop.tags if t.id is None]
+        if any(new_tags):
+            for t in new_tags:
+                db.session.add(t)
+            db.session.commit()
+
```

Comment:
Could you add a comment that this is done in a separate transaction to prevent races against other sessions, maybe pull it out into a separate function called something like `ensure_tags` that does an upsert? I can see that we don't want to blow up after sending emails.

Reply:
Realised that this is no longer needed. I'd added this as a fix for the uniqueness error I was seeing from `proposal_tag_version` when adding 2 new tags but that was actually because I wasn't defining a compound key to the `proposal_tag` table (so the `_version` was only paying attention to one of the `ids`)